compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/repro-missing-phi-after-dce-merge-scopes.js

```diff
@@ -0,0 +1,20 @@
+function Component() {
+  let v3, v4, acc;
+  v3 = false;
+  v4 = v3;
+  acc = v3;
+  if (acc) {
+    acc = true;
+    v3 = acc;
+  }
+  if (acc) {
+    v3 = v4;
+  }
+  v4 = v3;
+  return [acc, v3, v4];
+}
+
+export const FIXTURE_ENTRYPOINT = {
+  fn: Component,
+  params: [],
+};
```

Comment:
this is a great repro case, thank you @asmjmp0!

Reply:
<img width="1728" alt="Screenshot 2025-04-02 at 12 03 52 AM" src="https://github.com/user-attachments/assets/245c9b4a-3f13-4d5b-afbf-708f7bc30b30" />

Reply:
<img width="1728" alt="Screenshot 2025-04-02 at 12 03 52 AM" src="https://github.com/user-attachments/assets/5eda9f9a-ef33-40f9-9db5-9f93b7096162" />